lib/src/controls/navigation/tree_view.dart

```diff
@@ -716,9 +738,12 @@ class _TreeViewItem extends StatelessWidget {
                 child: Row(
                   crossAxisAlignment: CrossAxisAlignment.center,
                   children: [
+                    // Checkbox for multi selection mode
                     if (selectionMode == TreeViewSelectionMode.multiple)
                       Padding(
-                        padding: const EdgeInsetsDirectional.only(end: 20.0),
+                        padding: EdgeInsets.only(
+                          right: narrowSpacing ? 0.0 : _whiteSpace,
+                        ),
```

Comment:
We must take directionality into consideration as well
```suggestion
                        padding: EdgeInsetsDirectional.only(
                          end: narrowSpacing ? 0.0 : _whiteSpace,
                        ),
```